core/domain/collection_services.py

```diff
@@ -1287,17 +1271,12 @@ def index_collections_given_ids(collection_ids):
     """Adds the given collections to the search index.
 
     Args:
-        collection_ids: list(str). List of collection ids whose collections are
-            to be indexed.
+        collection_ids: list(str). List of collection ids whose collections
+        are to be indexed.
     """
-    # We pass 'strict=False' so as not to index deleted collections.
-    collection_list = get_multiple_collections_by_id(
-        collection_ids, strict=False).values()
-    search_services.add_documents_to_index([
-        _collection_to_search_dict(collection)
-        for collection in collection_list
-        if _should_index(collection)
-    ], SEARCH_INDEX_COLLECTIONS)
+    activity_search_services.index_collection_summaries([
+        get_collection_summary_by_id(collection_id) for
```

Comment:
You should use something like get_multiple_collections_by_id() here, otherwise you are making individual RPCs to the datastore and that's costly.